java/src/jmri/jmrit/logixng/expressions/ExpressionBlock.java

```diff
@@ -387,7 +400,9 @@ public boolean evaluate() throws JmriException {
             checkBlockState = BlockState.valueOf(getNewState());
         }
 
-        int currentState = _eventState;
+        int currentState = getBlock().getBean().getState();
```

Comment:
I recommend that you use `block` instead of `getBlock().getBean()` in the method `evaluate()`. `getBlock().getBean()` will work as long as the user uses direct addressing (selecting the block in the dialog), but will fail if the user uses indirect addressing, for example uses a reference or a formula to address the block.

For the method `isBlockAllocated()`, I recommend that you have the block as a parameter: `isBlockAllocated(Block block)`.

Reply:
The recommended changes have been applied.

However, due to a couple of commit issues, this branch is now useless.

Reply:
Can I help get the branch back to useful?

Here's the recent history:
![image](https://user-images.githubusercontent.com/2774117/115463399-c3de8180-a1f9-11eb-970d-63e5cd5e46bc.png)

Would it be possible to drop back a commit or two and preserve most of the work?

Reply:
@bobjacobsen 
Thanks, but it's not needed. The changes in this PR is now in PR #9707 that's ready to be merged after 24 hours.

Reply:
In this case, the changes were minor and in one class, so I made a new branch and it is waiting for the merge.  This PR needed some additional changes after I messed up the branch.

For future reference, it might be useful to know how to remove a commit from a branch.

38b0256 is the valid commit.  The next two occurred because I lost track of which branch I was using.  I was able to get them to the correct branch using `git cherry-pick SHA`.  The additional commits where a combination of command line attempts along with GitHub desktop.  

Based on the image, it looks like I was on the right track but then I reverted the revert for the Actions commit, but missed reverting Expressions.

BTW:  What program is the image from?